process/smartContract/process.go

```diff
@@ -444,14 +457,58 @@ func (sc *scProcessor) executeSmartContractCall(
 	return vmOutput, nil
 }
 
+func (sc *scProcessor) isSCROnlyInformational(scr data.TransactionHandler) bool {
+	if scr.GetValue().Cmp(zero) > 0 {
+		return false
+	}
+
+	function, _, err := sc.argsParser.ParseCallData(string(scr.GetData()))
+	if err != nil {
+		return true
+	}
+
+	if core.IsSmartContractAddress(scr.GetRcvAddr()) {
+		return false
+	}
+
+	_, err = sc.builtInFunctions.Get(function)
+	if err != nil {
```

Comment:
`return err!= nil` instead of `if { .. } .. `

Reply:
I would leave as it is. It is much cleaner